File: PJDesignBackStageClient/src/app/shared/models/get-units.ts
export class GetUnitsRequest {
  stageType?: number;
  groupId?: number;
  templateType?: number;

  constructor(stageType?: number, templateType?: number, groupId?: number) {
    this.stageType = stageType;
    this.templateType = templateType;
    this.groupId = groupId;
  }
}

export class GetUnitsResponse {
  id: number;
  name: string;
  backStageUrl?: string;
  templateType: number;
  frontStageUrl?: string;
  isAnotherWindow: boolean;
  isEnabled?: boolean;
  createDt: Date;
  parent?: number;
  stageType: number;
  sort?: number;
}

export class UnitList extends GetUnitsResponse {
  children: UnitList[];
}

File: PJDesignBackStageAPI/App.Model/Unit/GetUnits.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App.Model
{
    public class GetUnitsRequest
    {
        public int? StageType { get; set; }
        public int? GroupId { get; set; }
        public int? TemplateType { get; set; }
    }

    public class GetUnitsResponse
    {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string? BackStageUrl { get; set; }
        public int TemplateType { get; set; }
        public string? FrontStageUrl { get; set; }
        public bool IsAnotherWindow { get; set; }
        public bool? IsEnabled { get; set; }
        public DateTime CreateDt { get; set; }
        public int? Parent { get; set; }
        public int StageType { get; set; }
        public byte? Sort { get; set; }
    }
}
